### PR TITLE
Added VoidService and gateway#void

### PR DESCRIPTION
**Pull request by abhishekgupta5**
**Issue:** https://linear.app/nebulab-retainers/issue/ABU-10/implement-soliduspaytomorrowgatewayvoid

**Brief:**
- [x] Added VoidService in payTomorrow Client
- [x] Added specs for ^
- [x] Use the service in `gateway.rb`
- [x] Update gateway specs

### CODE REVIEW

File: app/services/solidus_pay_tomorrow/client/void_service.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module SolidusPayTomorrow
+  module Client
+    class VoidService < SolidusPayTomorrow::Client::BaseService
+      attr_reader :order_token
+
+      CANCEL_ENDPOINT = 'api/ecommerce/application/:order_token/cancel'
```
Comment (Naokimi) [inline app/services/solidus_pay_tomorrow/client/void_service.rb:8]: doesn't it need `.freeze` at the end?

File: spec/services/solidus_pay_tomorrow/client/void_service_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+require 'spec_helper'
+
+RSpec.describe SolidusPayTomorrow::Client::VoidService do
+  before do
+    ENV['API_BASE_URI'] = 'https://api-base-url.com'
+  end
+
+  describe '#call' do
+    it 'successful cancellation', vcr: { cassette_name: 'client/void/success' } do
+      response = described_class.call(order_token: 'order_token')
```
Comment (Naokimi) [inline spec/services/solidus_pay_tomorrow/client/void_service_spec.rb]: the service successfully canceled an order with a token `'order_token'`? 🤯

File: app/services/solidus_pay_tomorrow/client/void_service.rb (hunk 1 above)
Comment (abhishekgupta5) [reply app/services/solidus_pay_tomorrow/client/void_service.rb:8]: `frozen_string_literal: true` at the beginning of the file tells Ruby that all strings in the file are implicitly frozen.
Hence, in this case, freeze isn't needed for string types

File: spec/services/solidus_pay_tomorrow/client/void_service_spec.rb (hunk 2 above)
Comment (abhishekgupta5) [reply spec/services/solidus_pay_tomorrow/client/void_service_spec.rb]: No. The test passed with a valid token, but need not have it in version control, hence replaced a valid_token with `order_token` in yml and test.

Is this not the recommended way?

Comment (abhishekgupta5) [reply spec/services/solidus_pay_tomorrow/client/void_service_spec.rb]: Ignore this^ comment. We decided not to use VCR with this extension

File: spec/models/solidus_pay_tomorrow/gateway_spec.rb [hunk 3]
```diff
@@ -41,15 +41,38 @@
...
+      expect(result).to be_an_instance_of(ActiveMerchant::Billing::Response)
+      expect(result).to be_success
+    end
+
+    it 'returns an active merchant billing failure response' do
+      allow(SolidusPayTomorrow::Client::VoidService).to receive(:call).and_raise(StandardError)
+      result = described_class.new.void(payment.response_code,
+        originator: payment)
+      expect(result).to be_an_instance_of(ActiveMerchant::Billing::Response)
+      expect(result).not_to be_success
```
Comment (DanielePalombo) [inline spec/models/solidus_pay_tomorrow/gateway_spec.rb]: The 2 examples looks very strange.
Do you mind creating a different context for them?

Comment (abhishekgupta5) [reply spec/models/solidus_pay_tomorrow/gateway_spec.rb]: Added [here](https://github.com/nebulab/solidus_pay_tomorrow/pull/5/commits/ce1e367348f5d6f849f557f6c51a94ec6c50478e#diff-29658277569ec4cc1cf064679c583c4ee95610938fab468405b20eab45d82cb0)

**Review by DanielePalombo — APPROVED** [review]
LGTM

**Pull request merged**